### PR TITLE
Ransomware quickstart - Add get method for island mode

### LINKED ISSUE #1241: Ransomware quickstart
# Description

As a red team member, I want a quick and easy way to configure a ransomware simulation, so that I can easily test my network under a variety of ransomware scenarios.

# Acceptance Criteria
- A "Ransomware" button is added to the monkey island welcome screen.
- Upon clicking the ransomware button, the user is provided with a series of screens that allow them to:
    - enable/disable encryption
    - configure networking
    - configure exploits
- All post-breach actions are disabled
- All system info collectors are enabled
- Sidebar menu has a "Ransomware Simulation Configuration" item
- Testing plan is updated to include testing this UI.

# Tasks
- [x] Add API endpoint that allows the UI to set the mode (0d) - @VakarisZ 
    - A field is set in mongodb to store which mode the user chose
- [x] Add API endpoint that allows the UI to get the mode (0d) - @ilija-lazoroski 
- [x] Add a new welcome screen that is displayed when monkey first starts (0d) - @VakarisZ 
  - [x] No list items are displayed in sidebar
  - [x] A description of monkey is displayed (see current home page)
  - [x] 2 buttons allow the user to chose between Ransomware Simulation and Advanced Configuration
  - [x] When button is clicked
    - [x] Call API endpoint to set the mode
    - [x] User is brought to current home page
- [x] When the user clicks "start over", they are brought back to the ransomware/advanced configuration selection page (0d) @ilija-lazoroski @VakarisZ 
    - [x] API endpoint is called that unsets the mode
- [x] Set default configuration values when mode is set (potentially use config templates from blackbox tests) (0d) - @shreyamalviya 
    - [x] No change to behavior if mode is set to "Advanced"
    - [x] Disable post-breach actions if mode is set to "Ransomware"
- [x] Show mode at the top of the sidebar (0d) @VakarisZ 
- [x] Hide scoutsuite run option in ransomware mode (0d) @ilija-lazoroski 
- [x] Fix #1329  (0d)
- [x] Only display Exploits, Network, and Ransomware configuration tabs in Ransomware mode (0d) - @shreyamalviya 
- [x] Remove "Congratulations" message from island home page, change header to "Get started" (0d) @ilija-lazoroski

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do? 

Fixes #1241 get mode endpoint. 

Add any further explanations here. 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing? 
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [x] ~If applicable, add screenshots or log transcripts of the feature working~

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py [hunk 1]
```diff
@@ -54,3 +54,18 @@ def test_island_mode_post__set_invalid_model(flask_client, uses_database):
         "/api/island-mode", data=json.dumps({"mode": "bogus mode"}), follow_redirects=True
     )
     assert len(IslandMode.objects) == 0
+
+
+@pytest.mark.parametrize("mode", ["ransomware", "advanced"])
+def test_island_mode_get(flask_client, uses_database, mode):
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py:60]: this also tests the `post`. Maybe we don't need the `test_island_mode_post__set_model` and `test_island_mode_post__set_invalid_model` tests anymore?

Comment (shreyamalviya) [reply monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py:60]: Tests should be individual i.e. testing one kind of behaviour. You should set `IslandMode.objects` and then test just the post request.

**Review by shreyamalviya — CHANGES_REQUESTED** [review]

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py:60]: There's no need to make unit tests too granular. The result of the `post` is that `get` returns the mode posted. There's no need to tie the unit test to the mechanism of how the mode is saved. If we decide to switch the database we won't need to re-do unit tests, for example.

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Remove the UT's that interact with models directly and we'll be GTG

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py [hunk 2]
```diff
@@ -42,15 +42,16 @@ def test_island_mode_post__internal_server_error(monkeypatch, flask_client):
...
-def test_island_mode_post__set_model(flask_client, uses_database):
-    flask_client.post(
-        "/api/island-mode", data=json.dumps({"mode": "ransomware"}), follow_redirects=True
-    )
-    assert IslandMode.objects[0].mode == "ransomware"
+@pytest.mark.parametrize("mode", ["ransomware", "advanced"])
+def test_island_mode_endpoint(flask_client, uses_database, mode):
+    flask_client.post("/api/island-mode", data=json.dumps({"mode": mode}), follow_redirects=True)
+    resp = flask_client.get("/api/island-mode", follow_redirects=True)
+    assert json.loads(resp.data)["mode"] == mode
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py:49]: also check status here

**Review by VakarisZ — APPROVED** [review]

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py]: I don't agree. If the test fails, you're now left debugging more than if the tests were granular. 
I don't think our motive behind writing unit tests should be to not have to redo them ever again. It should be to make debugging and catching issues easier.

**Pull request merged**